pkg/gateway/model_build_rule.go

```diff
@@ -34,151 +35,154 @@ const (
 			}
 
 			// header based match
 			// today, only support EXACT match
-			if match.Headers != nil {
-				if len(match.Headers) > LATTICE_MAX_HEADER_MATCHES {
+			if match.GetHeaders() != nil {
+				if len(match.GetHeaders()) > LATTICE_MAX_HEADER_MATCHES {
 					return errors.New(LATTICE_EXCEED_MAX_HEADER_MATCHES)
 				}
 
-				ruleSpec.NumOfHeaderMatches = len(match.Headers)
+				ruleSpec.NumOfHeaderMatches = len(match.GetHeaders())
 
 				glog.V(6).Infof("Examing match.Headers %v for httproute %s namespace %s",
-					match.Headers, t.httpRoute.Name, t.httpRoute.Namespace)
+					match.GetHeaders(), t.route.GetName(), t.route.GetNamespace())
 
-				for i, header := range match.Headers {
-					glog.V(6).Infof("Examing match.Header: i = %d header.Type %v", i, *header.Type)
-					if header.Type != nil && gateway_api.HeaderMatchType(*header.Type) != gateway_api.HeaderMatchExact {
+				for i, header := range match.GetHeaders() {
+					glog.V(6).Infof("Examing match.Header: i = %d header.Type %v", i, *header.GetType())
+					if header.GetType() != nil && *header.GetType() != gateway_api.HeaderMatchExact {
 						glog.V(2).Infof("Unsupported header matchtype %v for httproute %v namespace %s",
-							*header.Type, t.httpRoute.Name, t.httpRoute.Namespace)
+							*header.GetType(), t.route.GetName(), t.route.GetNamespace())
 						return errors.New(LATTICE_UNSUPPORTED_HEADER_MATCH_TYPE)
 					}
 
 					glog.V(6).Infof("Found HeaderExactMatch==%v for HTTPRoute %v, namespace %v",
-						header.Value, t.httpRoute.Name, t.httpRoute.Namespace)
+						header.GetValue(), t.route.GetName(), t.route.GetNamespace())
 
 					matchType := vpclattice.HeaderMatchType{
-						Exact: aws.String(header.Value),
+						Exact: aws.String(header.GetValue()),
 					}
 					ruleSpec.MatchedHeaders[i].Match = &matchType
-					header_name := header.Name
+					header_name := header.GetName()
 
 					glog.V(6).Infof("Found matching i = %d header_name %v", i, &header_name)
 
-					ruleSpec.MatchedHeaders[i].Name = (*string)(&header_name)
+					ruleSpec.MatchedHeaders[i].Name = &header_name
 				}
 			}
 
-			// controller do not support these match type today
-			if match.Method != nil || match.QueryParams != nil {
-				glog.V(2).Infof("Unsupported Match Method %v for httproute %v, namespace %v",
-					match.Method, t.httpRoute.Name, t.httpRoute.Namespace)
-				return errors.New(LATTICE_UNSUPPORTED_MATCH_TYPE)
-			}
 			glog.V(6).Infof("Generated ruleSpec is: %v", ruleSpec)
 
 			tgList := []*latticemodel.RuleTargetGroup{}
 
-			for _, httpBackendRef := range httpRule.BackendRefs {
-				glog.V(6).Infof("buildRoutingPolicy - examing backendRef %v\n", httpBackendRef)
-				glog.V(6).Infof("backendref kind: %v\n", *httpBackendRef.BackendObjectReference.Kind)
+			for _, backendRef := range rule.GetBackendRefs() {
+				glog.V(6).Infof("buildRoutingPolicy - examining backendRef %v\n", backendRef)
+				glog.V(6).Infof("backendRef kind: %v\n", *backendRef.GetKind())
```

Comment:
boy scout rule: if you see something is off, dont leave it. Multiple log statements for single operation in general considered as log pollution, please squash them, it does not impact code behavior but makes code cleaner with every PR

Reply:
I fixed this one in next commit, but would like to avoid changes that are not directly related to this refactor.